src/qtils/bytes.hpp

```diff
@@ -11,17 +11,146 @@
 #include <span>
 #include <vector>
 
+#include <qtils/assert.hpp>
+
 namespace qtils {
   using Bytes = std::vector<uint8_t>;
   template <size_t N>
   using BytesN = std::array<uint8_t, N>;
   using BytesIn = std::span<const uint8_t>;
   using BytesOut = std::span<uint8_t>;
 
+  template <size_t N>
+  using ByteArray = std::array<uint8_t, N>;
+
+  using ByteSpan = std::span<const unsigned char>;
+
+  template <size_t N>
+  using FixedByteSpan = std::span<const unsigned char, N>;
+
+  template <size_t N>
+  using FixedByteSpanMut = std::span<unsigned char, N>;
+
+  using ByteSpanMut = std::span<unsigned char>;
+
   template <typename T>
   concept AsBytes = requires(const T &t) { BytesIn{t}; };
 
   inline Bytes asVec(BytesIn v) {
     return {v.begin(), v.end()};
   }
+
+  /**
+   * Fixed-capacity vector meant to be allocated on stack
+   * Used to avoid heap allocations for small vectors with known capacity
+   * @tparam N - maximum capacity
+   */
+  template <size_t N>
+  struct FixedByteVector {
```

Comment:
This type looks incomplete,
it's missing `size`/`push`/`append`/... to work with it.

Should it be added to qtils later when it's interface will be decided?

Reply:
It contains what was required of it. I see no harm in adding it now and adding some new stuff later if it's needed.